classes/AnyPage.php

```diff
@@ -32,8 +44,8 @@ public function save() {
 		$site = elgg_get_site_entity();
 
 		$this->access_id = ACCESS_PUBLIC;
-		$this->container_guid = $site->getGUID();
-		$this->owner_guid = $site->getGUID();
+		$this->container_guid = $site->guid;
+		$this->owner_guid = $site->guid;
```

Comment:
I don't agree with these points, especially about the "benefit" of avoiding fatal errors (you don't want to call properties on the wrong objects any more than calling methods on the wrong objects), but it's fine here.